### PR TITLE
Refactor tally

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

File: xprof/btx_tally.cpp [hunk 1]
```diff
@@ -10,6 +10,90 @@
...
+  static constexpr std::array headers{"Time", "Time(%)", "Calls", "Average", "Min", "Max", "Error"};
+  virtual const std::vector<std::string> to_string() {
+    return std::vector<std::string>{
+        (count == error) ? "" : this->format_time(duration),
+        (count == error) ? "" : this->to_pretty_string(100. * duration_ratio, "%"),
+        this->to_pretty_string(count, "", 0),
+        (count == error) ? "" : this->format_time(average()),
+        (count == error) ? "" : this->format_time(min),
+        (count == error) ? "" : this->format_time(max),
+        this->to_pretty_string(error, "", 0)};
```
Comment (TApplencourt) [inline xprof/btx_tally.cpp:25]: Should Fix #116

**Comment by TApplencourt** [comment]
Please don't merge, need more thinking.

**Comment by Kerilk** [comment]
Does it fix the trace Sarbojit provided?

File: hip/hipinterval_callbacks.cpp.erb [hunk 2]
```diff
@@ -101,7 +101,11 @@ static void hipinterval_<%= dbt_event.name %>_callback(
       <%elsif dbt_event.name_unsanitized.end_with?(STOP) %>
     uint64_t &_start_g = _state->host_start.at(hpt_function_name_t(_hostname, _process_id, _thread_id, "<%= dbt_event.name_striped  %>"));
       <% if dbt_event.fields_name.include?('hipResult') && dbt_event.fields_type['hipResult'] == 'hipError_t' %>
-    create_and_enqueue_host_message(_hostname.c_str(), _process_id, _thread_id, "<%= dbt_event.name_striped %>", _start_g, _ns_from_origin-_start_g, hipResult);
+     // Not an Error
+     if (hipResult == hipErrorNotReady)
```
Comment (Kerilk) [inline hip/hipinterval_callbacks.cpp.erb:105]:
```suggestion
    // Not an Error
    if (hipResult == hipErrorNotReady)
```

File: xprof/btx_tally.cpp (hunk 1 above)
Comment (Kerilk) [reply xprof/btx_tally.cpp:25]: Fixes #116

**Pull request merged**